### DISCUSSION TITLE
[Request] 我想请教两个问题 ，关于授权登录的

### DISCUSSION THREAD

**Discussion by anjiacm** [lobehub/lobehub#1333] [General Question] [answered]
### 🥰 需求描述

目前是不是不支持登录后再打开网站的功能呢
另外想请教下，我能否外置一个服务基于微信来调用lobechat的服务。因为我们员工可能会是用微信在下班期间需要使用查询公司数据这样

### 🧐 解决方案

无

[a comment by an automated account is not shown]

**Comment by cy948** [comment]
关于自定义身份验证方式的[wiki](https://github.com/lobehub/lobe-chat/wiki/Add-New-Authentication-Providers.zh-CN)可能对您有帮助。

**Comment by anjiacm** [comment]
外置一个服务基于微信来调用lobechat的服务  这个有办法吗

[a comment by an automated account is not shown]

**Comment by anjiacm** [comment]
> 关于自定义身份验证方式的[wiki](https://github.com/lobehub/lobe-chat/wiki/Add-New-Authentication-Providers.zh-CN)可能对您有帮助。

外置一个服务基于微信来调用lobechat的服务 这个有办法吗

**Comment by cy948** [comment, marked as answer]
> > 关于自定义身份验证方式的[wiki](https://github.com/lobehub/lobe-chat/wiki/Add-New-Authentication-Providers.zh-CN)可能对您有帮助。
> 
> 外置一个服务基于微信来调用lobechat的服务 这个有办法吗

暂时不清楚如何外置一个服务去调用lobechat的服务。Lobechat已在 #1143 中添加了单点登录功能，你可以通过使用提供单点登录的平台，如 Auth0, Authing 等实现对贵公司员工的身份验证，方式有很多：
1. 将注册人员的邮箱后缀限定为在贵公司的域名。
2. 使用企业微信的OAuth登陆。
3. 手动授权。

**Reply by SunLemuria** [reply]
你好，对这块不是很懂，请问有详细一点的教程吗，Auth0、Authing是否都需要将身份信息发送到第三方服务器呢，有没有不需发送到外部服务器的呢，可以以配置微信登录为例说明一下详细过程吗

**Reply by cy948** [reply]
1. 不需要发送身份信息到第三方服务的方法：配置自定义 Nextauth Provider， 如[email provider](https://authjs.dev/getting-started/authentication/email) 或自定义的 sso 到 [NextAuth Config](https://github.com/lobehub/lobe-chat/blob/main/src/libs/next-auth/index.ts) 中；
2. ”微信登录“的详细过程：目前，“微信登录“功能不在社区支持范围内，故暂无说明；

**Reply by SunLemuria** [reply]
企微也是在NextAuth Config中配置吗

**Reply by SunLemuria** [reply]
企微的OAuth需要第三方服务支持吗
email provider只能是[email provider](https://authjs.dev/getting-started/authentication/email)这里的几种之一吗，比如我想用企微邮箱，也是可以的吗

**Reply by cy948** [reply]
兼容 SMTP 协议即可。

[a comment by an automated account is not shown]
